packages/hash/shared/src/save.ts

```diff
@@ -483,13 +349,19 @@ const calculateSaveActions = async (
       beforeBlockDraftIds.splice(position, 0, afterDraftId);
     }
   }
+
+  actions.push(
+    ...moveActions.filter((action) => action.removeBlock),
+    ...moveActions.filter((action) => action.insertBlock),
+  );
```

Comment:
Why is this change necessary? Is this fixing a previously unknown bug?

Reply:
Yes, it's about dealing with the lack of atomicity. The backend does not play well with inserting a block at a position before removal